src/planwise/client/projects2/handlers.cljs

```diff
@@ -31,6 +39,23 @@
                      (assoc :list new-list))
       :navigate  (routes/projects2-show {:id project-id})})))
 
+(rf/reg-event-fx
+ :projects2/infer-step
+ in-projects2
+ (fn [{:keys [db]} [_ project]]
+   (let [steps ["goal", "consumers", "providers", "coverage", "actions", "review"]
+         first-invalid-step (first (filter #(not (s/valid? (keyword (str "planwise.model.project-" %) "validation") project)) steps))
+         selected-step (if (not-empty first-invalid-step) first-invalid-step "review")]
+     {:navigate (routes/projects2-show-with-step {:id (:id project) :step selected-step})})))
+
+(rf/reg-event-fx
+ :projects2/next-step-project
+ in-projects2
+ (fn [{:keys [db]} [_ project-id step]]
+   (let [steps ["goal", "consumers", "providers", "coverage", "actions", "review"]]
+     {:navigate (routes/projects2-show-with-step {:id project-id :step (steps (inc (.indexOf steps step)))})})))
```

Comment:
`.indexOf` is from JS. This will not work in general for CLJS structures. You can have the next step as part of the section description.

Reply:
Btw, since this is the third time you have repeated the vector of steps, it's screaming for refactor :) Clearly the list of steps has more entity by itself.